src/main/java/com/hazelcast/azure/AzureDiscoveryStrategyFactory.java

```diff
@@ -67,6 +67,13 @@ public DiscoveryStrategy newDiscoveryStrategy(DiscoveryNode node, ILogger logger
      * @return {@code Collection<PropertyDefinition>} the property defitions for the AzureDiscoveryStrategy
      */
     public Collection<PropertyDefinition> getConfigurationProperties() {
-        return REQUIRED_PROPERTY_DEFINITIONS;
+        List<PropertyDefinition> propertyDefinitions = new ArrayList<PropertyDefinition>();
+        propertyDefinitions.add(AzureProperties.CLIENT_ID);
+        propertyDefinitions.add(AzureProperties.CLIENT_SECRET);
+        propertyDefinitions.add(AzureProperties.TENANT_ID);
+        propertyDefinitions.add(AzureProperties.SUBSCRIPTION_ID);
+        propertyDefinitions.add(AzureProperties.GROUP_NAME);
+        propertyDefinitions.add(AzureProperties.CLUSTER_ID);
+        return propertyDefinitions;
```

Comment:
I didn't make it immutable list because it creates a new list everytime it's called. Do you think it worth the added protection?

Reply:
Think of it as presenting the correct invariants regardless of how the method is implemented. Even if we create a new list on every call, callers of this method should not assume they can modify the returned list. While doing so, with this implementation, won't cause an issues, since any changes won't propagate, it sets the wrong precedent (and means that if you ever switched over to a constant-based approach, such callers would break). It's better to have the returned value function consistent with how you expect it to be used.